src/AzurePipelineParser/Models/Resources.cs

```diff
@@ -8,15 +8,15 @@ public class Resources : IPipelineNode
     {
         private List<IPipelineNode> _children = new List<IPipelineNode>();
 
-        public PipelineResource? Pipelines { get; set; } 
+        public IList<PipelineResource> Pipelines { get; set; } = new List<PipelineResource>();
```

Comment:
Usually in a tree representation these would be under a `PipelineResources` node that contain the list of pipelines as it's children, no? Same for all the others that you are changing to an IList.